index.src.html

```diff
@@ -62,51 +62,94 @@
-example, web authors may call `decode()` without waiting for the previous
-`decode()` to generate an output. This is facilitated by the following
-mechanisms.
+Background {#processing-model-background}
+-----------------------------------------
 
-Each codec has a single <dfn>control message queue</dfn> that is a <a>queue</a>
-of <dfn>control messages</dfn>.
+This section is non-normative.
 
-<dfn lt="Enqueues a control message|Queue a control message">Queuing a control
-    message</dfn> means <a>enqueing</a> the message to the codec’s <a>control
-    message queue</a>. Invoking codec methods will often queue a control message
-    to schedule work.
+The codec interfaces described herein are designed such that new codec tasks may
+be scheduled while previous tasks are still pending. For example, web authors
+may call `decode()` without waiting for a previous `decode()` to complete.
+This is achieved by offloading underlying codec tasks to a separate thread for
+parallel execution.
 
-<dfn lt="running a control message|control message steps">Running a control
-    message</dfn> means executing a sequence of steps specified by the method
-    that enqueued the message.
+This section describes threading behaviors as they are visible from the
+perspective of web authors. Implementers may choose to use more or less threads
+as long the exernally visible behaviors of blocking and sequencing are
+maintained as follows.
+
+Control Thread and Codec Thread {#control-thread-and-codec-thread}
+------------------------------------------------------------------
 
-<dfn lt="Run the control message processing loop">Running the control message
-    processing loop</dfn> means executing these steps.
-1. While the control message queue is not <a>empty</a>
-    1. Let |front message| be the next <a>dequeued</a> <a>control message</a>
-    2. If |front message| cannot be executed now, return.
+All steps in this specificaiton will run on either a [=control thread=] or
+a [=codec thread=].
 
-        The User Agent must decide when further processing is blocked because of
-            ongoing work as an implementation detail (e.g. the underlying
-            decoder cannot accept more requests yet). The UA must restart the
-            processing loop when the blockage is resolved.
+The <dfn>control thread</dfn> is the thread from which authors will construct
+and [=codec=] and invoke its methods. Invoking a codec's methods will typically
+result in the creation of [=control messages=] which are later executed on the
+[=codec thread=]. Each [=JavaScript realm=] has a separate control thread.
 
-        NOTE: a blocked processing loop is visible to authors via the
-            `decodeQueueSize` and `encodeQueueSize` attributes.
+The <dfn>codec thread</dfn> is the thread from which a [=codec=] will
+[=dequeue=] [=control messages=] and execute their steps. Each [=codec=]
+instance has a separate codec thread.
```

Comment:
I mention here that its associated with a codec instance, but I don't have any steps that say "start the thread" or "stop the thread" as codec's are constructed/closed. Should those steps be added? Any pointers on phrasing?

Reply:
We can say it exists as long as the codec instance exists, since it's not useful to start it late or stop it early, even though an implementation might choose to do so, of course, and will be able to, or otherwise do something completely different, as we discussed yesterday.